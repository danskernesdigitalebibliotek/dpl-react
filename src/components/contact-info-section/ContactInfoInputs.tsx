import clsx from "clsx";
import * as React from "react";
import { FC } from "react";

export interface ContactInfoInputsProps {
  isInline: boolean;
  children: React.ReactNode;
  dataCy?: string;
  className?: string;
}

// This component wraps the input fields for the contact info section
// depending on the isInline prop.
const ContactInfoInputs: FC<ContactInfoInputsProps> = ({
  isInline,
  children,
  dataCy = "contact-info-input",
  className = undefined
}) => {
  if (!isInline) {
    return (
      <div className={className} data-cy={dataCy}>
        {children}
      </div>
    );
  }

  const renderableChildren = React.Children.toArray(children);
  return (
    <div
      className={clsx(className, {
        "dpl-input__flex": isInline
      })}
      data-cy={dataCy}
    >
      {renderableChildren.map((child, i) => {
        const childClassName = clsx("patron__input--desktop", {
          "mr-16": i < renderableChildren.length - 1
        });
        return (
          // eslint-disable-next-line react/no-array-index-key
          <div key={i} className={childClassName}>
            {child}
          </div>
        );
      })}
    </div>
  );
};

export default ContactInfoInputs;
