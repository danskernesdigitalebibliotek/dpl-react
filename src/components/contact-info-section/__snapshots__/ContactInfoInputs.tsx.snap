// Vitest Snapshot v1

exports[`ContactInfoInputs > Should NOT wrap the input fields if it is NOT inline 1`] = `
<div
  data-cy="contact-info-input-inline"
>
  <p>
    One input component
  </p>
  <p>
    Another input component
  </p>
</div>
`;

exports[`ContactInfoInputs > Should wrap the input fields if it is inline 1`] = `
<div
  class="dpl-input__flex"
  data-cy="contact-info-input"
>
  <div
    class="patron__input--desktop mr-16"
  >
    <p>
      One input component
    </p>
  </div>
  <div
    class="patron__input--desktop"
  >
    <p>
      Another input component
    </p>
  </div>
</div>
`;
